src/apps/competitions/views.py

```diff
@@ -20,6 +21,18 @@ class CompetitionDetail(LoginRequiredMixin, DetailView):
     queryset = Competition.objects.all()
     template_name = 'competitions/detail.html'
 
+    def get_object(self, *args, **kwargs):
+        competition = super().get_object(*args, **kwargs)
+
+        if self.request.user.is_superuser or self.request.user == competition.created_by or self.request.user in competition.collaborators.all():
```

Comment:
Hmm, this feels weird, what do you think of this style?

```python
is_creator = self.request.user.is_superuser or self.request.user == competition.created_by
is_collaborator = self.request.user in competition.collaborators.all()
valid_secret_key = self.request.GET.get('secret_key') == str(competition.secret_key)
if is_creator or is_collaborator or competition.published or valid_secret_key:
    return competition
raise Http404()
```

Reply:
- [x] done